### PR TITLE
[ISSUE #3140]nacos-client module replace http client

### PR DESCRIPTION
**Pull request by Maijh97**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

nacos-client module replace http client

## Brief changelog

fix: #3140 refer: #2858
- Unified use of httpclient

- use `nacosRestTemplate` replace `nacos-client` module `httpClient` 

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [x] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [x] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Comment by Maijh97** [comment]
The nacos-client module http client has been replaced by NacosRestTemplate,please help to review this pr, thanks. @chuntaojun @KomachiSion

File: common/src/main/java/com/alibaba/nacos/common/http/handler/ResponseHandler.java [hunk 1]
```diff
@@ -78,14 +78,22 @@
         String contentType = headers.getValue(HttpHeaderConsts.CONTENT_TYPE);
         InputStream body = response.getBody();
         T extractBody = null;
+        boolean typeToStr = String.class.toString().equals(type.toString());
```
Comment (KomachiSion) [inline common/src/main/java/com/alibaba/nacos/common/http/handler/ResponseHandler.java:81]: If the tmp variable is only used once, I suggest do not change.

File: common/src/main/java/com/alibaba/nacos/common/http/handler/ResponseHandler.java [hunk 2]
```diff
@@ -78,14 +78,22 @@
...
+            // then it should be serialized as string
+            if (typeToStr) {
+                extractBody = (T) IoUtils.toString(body, headers.getCharset());
+            } else {
+                extractBody = convert(body, type);
+            }
         }
         if (extractBody == null) {
-            if (!String.class.toString().equals(type.toString())) {
+            if (!typeToStr) {
```
Comment (KomachiSion) [inline common/src/main/java/com/alibaba/nacos/common/http/handler/ResponseHandler.java:93]: If the tmp variable is only used once, I suggest do not change.

File: common/src/main/java/com/alibaba/nacos/common/http/client/NacosRestTemplate.java [hunk 3]
```diff
@@ -298,7 +297,29 @@ public NacosRestTemplate(HttpClientRequest requestClient) {
...
+     * @param url          url
+     * @param header       http header param
+     * @param paramValues  http query param
+     * @param bodyValues   http body param
+     * @param httpMethod   http method
+     * @param responseType return type
+     * @return {@link HttpRestResult}
+     * @throws Exception ex
+     */
+    public <T> HttpRestResult<T> exchangeFrom(String url, Header header,
```
Comment (KomachiSion) [inline common/src/main/java/com/alibaba/nacos/common/http/client/NacosRestTemplate.java:314]: What's the mean of `From` I read the method think it should be `exchangeForm`, is it?

File: test/src/test/java/com/alibaba/nacos/test/naming/NamingBase.java [hunk 4]
```diff
@@ -166,39 +170,32 @@ public static boolean verifyInstanceList(List<Instance> instanceList1, List<Inst
...
         String url = "http://127.0.0.1:" + localPort + "/nacos/v1/ns/operator/switches?entry=autoChangeHealthCheckEnabled&value=" + true;
-        List<String> headers = new ArrayList<String>();
-        headers.add(HttpHeaderConsts.USER_AGENT_HEADER);
-        headers.add("Nacos-Server");
-        HttpClient.HttpResult result =
-            HttpClient.request(url, headers, new HashMap<String, String>(), StringUtils.EMPTY, "UTF-8", "PUT");
+        Header header = Header.newInstance();
+        header.addParam(HttpHeaderConsts.USER_AGENT_HEADER, "Nacos-Server");
+
+        HttpRestResult<String> result = nacosRestTemplate.putFrom(url, header, Query.EMPTY, new HashMap<>(), String.class);
```
Comment (KomachiSion) [inline test/src/test/java/com/alibaba/nacos/test/naming/NamingBase.java:197]: What's the mean of From I read the method think it should be `putFrom`, is it?

**Comment by Maijh97** [comment]
> Great implement!
> Only some style problem.

I have adjusted the code.please help to review this pr again, thanks.

**Pull request merged**